source/Halibut/ServiceModel/ServiceInvoker.cs

```diff
@@ -36,9 +38,111 @@ public ResponseMessage Invoke(RequestMessage requestMessage)
             }
         }
 
+        public async Task<ResponseMessage> InvokeAsync(RequestMessage requestMessage)
+        {
+            using var lease = factory.CreateService(requestMessage.ServiceName);
+            var methods = lease.Service.GetType().GetHalibutServiceMethods().Where(m => string.Equals(m.Name, requestMessage.MethodName, StringComparison.OrdinalIgnoreCase)).ToList();
```

Comment:
Should the lookup be the reverse? specifically prefer an async method over a sync method?

Reply:
Lets switch the order since its probably more likely that async code calling async code is a better default over async code calling sync code.

As a library maintainer, changing the order later could be hard since we don't know if someone has started to depend on the sync being called over async. Which is possible since we don't enforce either sync or async services.

Reply:
Makes sense, done.